### PR TITLE
Add Ruff linter pre-commit hook

### PR DESCRIPTION
**Pull request by sbrunner**
## Description

See: https://docs.astral.sh/ruff/integrations/#pre-commit

### CODE REVIEW

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
We should activate most of the check in particular UP and I in pyproject.toml (then remove isort and flake8)

**Comment by sbrunner** [comment]
Update, I also remove pyupgrade :-)

File: prospector/pathutils.py [hunk 1]
```diff
@@ -37,8 +33,4 @@ def is_virtualenv(path: Path) -> bool:
     # if we do have all three directories, make sure that it's not
     # just a coincidence by doing some heuristics on the rest of
     # the directory
-    if len(dircontents) > 7:
-        # if there are more than 7 things it's probably not a virtualenvironment
-        return False
-
-    return True
+    return len(dircontents) <= 7
```
Comment (Pierre-Sassoulas) [inline prospector/pathutils.py:36]:
```suggestion
    # if there are more than 7 things it's probably not a virtualenvironment
    return len(dircontents) <= 7
```

File: prospector/autodetect.py [hunk 2]
```diff
@@ -86,10 +87,8 @@ def autodetect_libraries(path: Union[str, Path]) -> set[str]:
 
     libraries: set[str] = set()
 
-    try:
+    with contextlib.suppress(RequirementsNotFound):
         libraries = find_from_requirements(path)
-    except RequirementsNotFound:
-        pass
```
Comment (Pierre-Sassoulas) [inline prospector/autodetect.py:92]: This is slower (I don't remember how much but noticeably) and is unusual imo we c/should skip this one.

**Review by Pierre-Sassoulas — COMMENTED** [review]
Great ! We can also use ruff-fmt instead of black

**Comment by sbrunner** [comment]
Thanks for the review, fixed :-)

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**